src/software/ai/hl/stp/tactic/goalie/simulated_goalie_tactic_test.cpp

```diff
@@ -0,0 +1,349 @@
+            {
+                yield("Tactic not done");
+            }
+            ballKicked((world_ptr->ball().position() - field().friendlyGoalCenter())
+                           .orientation(),
+                       world_ptr, yield);
+            checkGoalieSuccess(1, world_ptr, yield);
+        }};
+
+    std::vector<ValidationFunction> non_terminating_validation_functions = {
+        [](std::shared_ptr<World> world_ptr, ValidationCoroutine::push_type& yield) {
+            enemyNeverScores(world_ptr, yield);
+        }};
+
+    runTest(terminating_validation_functions, non_terminating_validation_functions,
+            Duration::fromSeconds(10));
+}
+
+// does not complete tactic because never leaves position_to_block state
+TEST_F(SimulatedGoalieTacticTest, test_ball_very_fast_misses_net)
+{
+    setBallState(BallState(Point(0, 0), Vector(-4, 1)));
+    addFriendlyRobots(TestUtil::createStationaryRobotStatesWithId({Point(-4.5, 0)}));
+
+    std::shared_ptr<const GoalieTacticConfig> goalie_tactic_config =
+        std::make_shared<const GoalieTacticConfig>();
+    auto tactic = std::make_shared<GoalieTactic>(goalie_tactic_config);
+    setTactic(tactic);
+    setRobotId(0);
+
+    std::vector<ValidationFunction> terminating_validation_functions = {
+        [this, tactic](std::shared_ptr<World> world_ptr,
+                       ValidationCoroutine::push_type& yield) {
+            while (!tactic->done())
+            {
+                yield("Tactic not done");
+                // break from loop since tactic will never complete
+                break;
+            }
+            checkGoalieSuccess(5, world_ptr, yield);
+        }};
+
+    std::vector<ValidationFunction> non_terminating_validation_functions = {
+        [](std::shared_ptr<World> world_ptr, ValidationCoroutine::push_type& yield) {
+            enemyNeverScores(world_ptr, yield);
+        }};
+
+    runTest(terminating_validation_functions, non_terminating_validation_functions,
+            Duration::fromSeconds(10));
+}
+
+// does not complete tactic because never leaves position_to_block state
+TEST_F(SimulatedGoalieTacticTest, test_slow_ball_at_sharp_angle_to_friendly_goal)
+{
+    setBallState(BallState(Point(-4.5, -3), Vector(0, 0.1)));
+    addFriendlyRobots(TestUtil::createStationaryRobotStatesWithId({Point(-4.5, 0)}));
+
+    std::shared_ptr<const GoalieTacticConfig> goalie_tactic_config =
+        std::make_shared<const GoalieTacticConfig>();
+    auto tactic = std::make_shared<GoalieTactic>(goalie_tactic_config);
+    setTactic(tactic);
+    setRobotId(0);
+
+    std::vector<ValidationFunction> terminating_validation_functions = {
+        [this, tactic](std::shared_ptr<World> world_ptr,
+                       ValidationCoroutine::push_type& yield) {
+            while (!tactic->done())
+            {
+                yield("Tactic not done");
+                // break from loop since tactic will never complete
+                break;
+            }
```

Comment:
When I first made the states/transitions in the fsm, it didn't really make sense for the position_to_block state to ever transition directly to X, the tactic should only be complete when danger has passed or the goalie cleared the ball - position_to_block_s is more just waiting for something to happen. The purpose of the last two test cases are to check that the goalie tracks the ball correctly, but it never transitions out of the position_to_block state, which means that the tactic will never be complete for the duration of the test.. I couldn't remove the loop altogether so that's why the break statement is there. That was the best I could come up with at the moment, but I'm definitely open to suggestions

Reply:
so what happens without the 
```
 while (!tactic->done())
            {
                yield("Tactic not done");
                // break from loop since tactic will never complete
                break;
            }
```
loop?

Reply:
The last time I tried to take the loop out, it gave me a weird error, but I just did it again and it was fine, I think the first time I must have forgotten to remove "tactic" from the arguments passed into the validation functions - so I'll just take the loop out then :)

Reply:
removed the `while (!tactic.done())` loops